File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.kt
fun main() {
    val removedNums = readln()
    var searchIndex = 0
    var num = 0

    while (searchIndex < removedNums.length) {
        num++
        val currentNum = num.toString()

        var i = 0  // current_num의 인덱스
        while (i < currentNum.length && searchIndex < removedNums.length) {
            if (currentNum[i] == removedNums[searchIndex]) {
                searchIndex++
            }
            i++
        }
    }

    println(num)
}

File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.py
from collections import deque
removed_nums = deque(input())

num = 0
while removed_nums:
    num+=1
    compare_nums = list(str(num))

    if removed_nums[0] in compare_nums:
        compare_nums = compare_nums[compare_nums.index(removed_nums[0]):]
        for c in compare_nums:
            if removed_nums and c == removed_nums[0]:
                removed_nums.popleft()
print(num)
